src/Microsoft.Diagnostics.Monitoring.EventPipe/Counters/CounterFilter.cs

```diff
@@ -33,8 +36,12 @@ public void AddFilter(string providerName, string[] counters)
 
         public IEnumerable<string> GetProviders() => _enabledCounters.Keys;
 
-        public bool IsIncluded(string providerName, string counterName)
+        public bool IsIncluded(string providerName, string counterName, int interval)
         {
+            if (_intervalMilliseconds != interval)
+            {
+                return false;
+            }
```

Comment:
> Is the intent to only listen if the interval matches what was requested?

Yes. This is mostly a move of the existing code from `EventCounterPipeline.OnEventSourceAvailable` to `CounterFilter` and the new `TraceEventExtensions.TryGetCounterPayload` method. You don't ever want to observe all counter events for a given counter because the counter value for a shorter interval (one that you might not have requested) may not be at the expected level whereas a longer interval (one that you did request) may be at the expected level. For example, cpu usage on short intervals _might_ have lower averages for some of the intervals whereas when measured for a longer interval the averages could be higher.